src/main/kotlin/no/nav/familie/ba/sak/ekstern/pensjon/PensjonService.kt

```diff
@@ -0,0 +1,102 @@
+import no.nav.familie.ba.sak.kjerne.beregning.domene.TilkjentYtelseRepository
+import no.nav.familie.ba.sak.kjerne.fagsak.FagsakRepository
+import no.nav.familie.ba.sak.kjerne.personident.PersonidentService
+import no.nav.familie.ba.sak.kjerne.vedtak.VedtakService
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+
+@Service
+class PensjonService(
+    private val behandlingHentOgPersisterService: BehandlingHentOgPersisterService,
+    private val fagsakRepository: FagsakRepository,
+    private val andelTilkjentYtelseRepository: AndelTilkjentYtelseRepository,
+    private val personidentService: PersonidentService,
+    private val vedtakService: VedtakService,
+    private val tilkjentYtelseRepository: TilkjentYtelseRepository
+) {
+    fun hentBarnetrygd(personIdent: String, fraDato: LocalDate): List<BarnetrygdTilPensjon> {
+        val barnetrygdTilPensjon = hentBarnetrygdTilPensjon(personIdent, fraDato)
+
+        // Sjekk om det finnes relaterte saker, dvs om barna finnes i andre behandlinger
+
+        val barnetrygdMedRelaterteSaker = barnetrygdTilPensjon?.barnetrygdPerioder
+            ?.filter { it.personIdent != barnetrygdTilPensjon.fagsakEiersIdent }
+            ?.map { barnetrygdPeriode -> hentBarnetrygdForRelatertPersonTilPensjon(barnetrygdPeriode.personIdent, fraDato) }
+            ?.flatten()
+        return barnetrygdMedRelaterteSaker?.plus(barnetrygdTilPensjon)
+            ?.toSet()?.toList() ?: emptyList()
+    }
+    private fun hentBarnetrygdForRelatertPersonTilPensjon(personIdent: String, fraDato: LocalDate): List<BarnetrygdTilPensjon> {
+        val aktør = personidentService.hentAktør(personIdent)
+        val andeler = andelTilkjentYtelseRepository.finnAndelerTilkjentYtelseForAktør(aktør)
+        // finn alle behandlinger
+        val behandlinger = andeler.map { it.behandlingId }.toSet()
+        // finn alle fagsaker
+        val aktørerTilEierAvFagsaker = behandlinger.map { behandlingHentOgPersisterService.hent(it).fagsak.aktør }
+        // hent alle barnetrygdperiodene for de aktuelle fagsakene.
+        return aktørerTilEierAvFagsaker.map { aktør -> hentBarnetrygdTilPensjon(aktør.aktivFødselsnummer(), fraDato) }.filterNotNull()
+    }
+    private fun hentBarnetrygdTilPensjon(personIdent: String, fraDato: LocalDate): BarnetrygdTilPensjon? {
+        val aktør = personidentService.hentAktør(personIdent)
+        val fagsak = fagsakRepository.finnFagsakForAktør(aktør)
+        val behandling = fagsak?.let { behandlingHentOgPersisterService.hentSisteBehandlingSomErIverksatt(it.id) }
+        if (fagsak == null || behandling == null) {
+            return null
+        }
+        logger.info("Henter perioder med barnetrygd til pensjon for fagsakId=${fagsak.id}, behandlingId=${behandling.id}")
+
+        val vedtak = vedtakService.hentAktivForBehandling(behandling.id)
+
+        var datoVedtak = vedtak?.vedtaksdato
```

Comment:
Disse 2 er ikke brukt nå
```suggestion
        val vedtak = vedtakService.hentAktivForBehandling(behandling.id)

        var datoVedtak = vedtak?.vedtaksdato
```